Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -1156,17 +1156,34 @@ def appendContext(key, data, dedup=False):
     if data is None:
         return
     existing = demisto.get(demisto.context(), key)
+
     if existing:
-        strBased = isinstance(data, STRING_TYPES) and isinstance(existing, STRING_TYPES)
-        if strBased:
-            data = data.split(',')
-            existing = existing.split(',')
-        newVal = data + existing
+        if isinstance(existing, STRING_TYPES):
+            if isinstance(data, STRING_TYPES):
+                new_val = data + ',' + existing
+            else:
+                return_error("Cannot append data to the existing context - \n The data is of type {} while the "
+                             "context in the specified path is of instance {}.".format(type(data), type(existing)))
+
+        if isinstance(existing, dict):
+            if isinstance(data, dict):
+                existing.update(data)
```

Comment:
I think this is not the expected behaviour.
If we have a 2 dictionaries, then I would expect to have list of 2 dictionaries in it.

Reply:
Changed to a list of two dicts